src/media/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { MicrophoneConstraints } from './local-audio-track';
+import { VideoConstraints } from './local-video-track';
+
+// For reference https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints
+/**
+ * Generates the audio constraints pass to getUserMedia.
+ *
+ * @param audioConstraints - Audio constraints.
+ * @returns Native audio constraints.
+ */
+export const generateAudioConstraints = (
+  audioConstraints?: MicrophoneConstraints
+): MediaTrackConstraints => {
+  let audio: MediaTrackConstraints = {};
+  if (audioConstraints?.microphoneDeviceId) {
+    audio.deviceId = audioConstraints.microphoneDeviceId;
+  }
+  if (audioConstraints?.echoCancellation) {
+    audio.echoCancellation = audioConstraints.echoCancellation;
+  }
+
+  if (audioConstraints?.noiseSuppression) {
+    audio.noiseSuppression = audioConstraints.noiseSuppression;
+  }
+
+  if (audioConstraints?.autoGainControl) {
+    audio.autoGainControl = audioConstraints.autoGainControl;
+  }
+
+  if (audioConstraints?.encoderConfig) {
+    audio = { ...audio, ...audioConstraints.encoderConfig };
+  }
+  return audio;
+};
+
+/**
+ * Generates the audio constraints pass to getUserMedia.
+ *
+ * @param videoConstraints - Video constraints.
+ * @returns Native video constraints.
+ */
+export const generateVideoConstraints = (
+  videoConstraints?: VideoConstraints
```

Comment:
the extra `VideoConstraints` type doesn't really seem to add much value, if we really want a separate type i'd use a `Partial` of the `MediaTrackConstraints` for video and audio

Reply:
it wont work because we are giving custom names to them microphoneDeviceId, cameraDeviceId, encoderConfig

Reply:
In my opinion those names don't add value that make it worthwhile to introduce the burden of maintaining the extra types.  I don't think `webrtc-core` needs to strive to be friendly to people who aren't familiar with webrtc details, instead it should just be convenient for those who are familiar to accomplish common tasks.

Reply:
In that case the meetings SDK will need to add these type in there code for the developers 

just wanted to avoid adding those types in calling and meetings SDK rather do it here we can change it if you are ok with it

Reply:
let me check with marcin on this and get back

Reply:
moved to start using the pick type